hcl2template/types.packer_config.go

```diff
@@ -280,13 +285,89 @@ func (cfg *PackerConfig) getCoreBuildPostProcessors(source SourceBlock, blocks [
 	return res, diags
 }
 
+type provisionerPrepare struct {
+	cfg               *PackerConfig
+	provisionersBlock []*ProvisionerBlock
+	src               SourceRef
+}
+
+// HCL2ProvisionerPrepare is used by the ProvisionHook at the runtime in the provision step
+// to interpolate any build variable by decoding and preparing it again.
+func (pp *provisionerPrepare) HCL2ProvisionerPrepare(data map[string]interface{}) ([]packer.CoreBuildProvisioner, hcl.Diagnostics) {
+	src := pp.cfg.Sources[pp.src.Ref()]
+
+	variables := make(Variables)
+	for k, v := range data {
+		if value, ok := v.(string); ok {
```

Comment:
is it possible that we get and integer here ? I specifically thinking of `PackerHTTPPort`

Reply:
No. This is always called by the `step_provision` after transforming all these variables into strings [here](https://github.com/hashicorp/packer/blob/master/common/step_provision.go#L30).

https://github.com/hashicorp/packer/blob/master/common/step_provision.go#L113 the hook is called in line 128 after that.

Reply:
Right ! Thanks